### DISCUSSION TITLE
[Unstable] "PirateMission: Enemy sub not created"

### DISCUSSION THREAD

**Discussion by Ek-Videogames** [FakeFishGames/Barotrauma#14681] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The pirate vessel never seems to spawn, and there's not a sonar marker either.

This was in the Aphotic Plateau "49% difficulty".

> [Vanilla] Error in PirateMission: enemy sub was not created.
![image](https://github.com/user-attachments/assets/e699723b-bd67-49ed-9801-9a6be28f21aa)

No other error information seems to show up, that's basically it good luck.

### Reproduction steps

1. Get a pirate mission and then load the map?
2. This error message might happen?

Could maybe be linked to alien ruins also spawning in the same map? But more likely that is just coincidence.

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Single player

### Version

v1.6.4.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by Rokvach** [comment]
Fixed and tested the fix to be working correctly, fix coming in the next Unstable build. Closing discussion.
